python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -156,9 +170,9 @@ def __init__(
         reader_and_node_list: List[Tuple["ray.actor.ActorHandle", str]],
         typ: Optional[Union[int, SharedMemoryType]] = None,
         _writer_node_id: Optional["ray.NodeID"] = None,
-        _reader_node_id: Optional["ray.NodeID"] = None,
+        _reader_node_ids: Optional[Set["ray.NodeID"]] = None,
         _writer_ref: Optional["ray.ObjectRef"] = None,
-        _reader_ref: Optional["ray.ObjectRef"] = None,
+        _node_id_to_reader_info: Optional[Dict[str, ReaderInfo]] = None,
```

Comment:
Is the keys the same as `_reader_node_ids`? Should we remove `_reader_node_ids`?

Does it contain the duplicate info as `reader_and_node_list`?

Reply:
it contains subset of `reader_and_node_list`. `reader_and_node_list ` contains all reader info, this one only contains the one that holds ref (1 per node)

For `_reader_node_ids ` good catch! it is removed

Reply:
SG. Can you clarify in the docstring?